app/controllers/api/base_controller.rb

```diff
@@ -27,8 +27,8 @@ def authenticate
       true
     end
 
-    def current_organization(api_key = nil)
-      @current_organization ||= Organization.find_by(api_key:)
+    def current_organization(api_key_value = nil)
+      @current_organization ||= ApiKey.find_by(value: api_key_value)&.organization
```

Comment:
Had a chat with Vincent about this. We agreed that it could be done but there’s no big reason for it because with the next step for api key management feature (rotating a key) it might create more confusion then benefits. Since old key shouldn't give access anymore.